### PR TITLE
chore(sdk-coin-sui): update bcs and transfer tx serialization

### PR DESCRIPTION
**Pull request by linyah9912**
Ticket: BOS-965

- update BCS
- update transfer transaction types, serialization/deserialization (pay, paySui, payAllSui)
- update staking transaction types

### CODE REVIEW

**Review by ewangbitgo — DISMISSED** [review]

**Review by margueriteblair — APPROVED** [review]

**Review by noel-bitgo — DISMISSED** [review]

File: modules/sdk-coin-sui/src/lib/constants.ts [hunk 1]
```diff
@@ -3,9 +3,9 @@ import { SharedObjectRef } from './iface';
 export const UNAVAILABLE_TEXT = 'UNAVAILABLE';
 export const TRANSFER_AMOUNT_UNKNOWN_TEXT = 'TRANSFER_AMOUNT_UNKNOWN';
 
-// Need to keep in sync with
-// https://github.com/MystenLabs/sui/blob/f32877f2e40d35a008710c232e49b57aab886462/crates/sui-types/src/messages.rs#L338
-export const SUI_GAS_PRICE = 1;
```
Comment (sachushaji) [inline modules/sdk-coin-sui/src/lib/constants.ts:8]: Seems to be a breaking change

File: modules/sdk-coin-sui/src/lib/iface.ts [hunk 2]
```diff
@@ -118,16 +118,21 @@ export interface MoveCallTxDetails {
...
  * The transaction data returned from the toJson() function of a transaction
  */
 export interface TxData {
   id?: string;
   kind: { Single: TxDetails };
   sender: string;
-  gasPayment: SuiObjectRef;
-  gasBudget: number;
-  gasPrice: number;
+  gasData: GasData;
```
Comment (sachushaji) [inline modules/sdk-coin-sui/src/lib/iface.ts:135]: Seems to be a breaking change

File: modules/sdk-coin-sui/src/lib/stakingBuilder.ts [hunk 3]
```diff
@@ -43,9 +43,7 @@ export class StakingBuilder extends TransactionBuilder<MoveCallTx> {
       type: SuiTransactionType.AddDelegation,
       sender: this._sender,
       tx: this._moveCallTx,
-      gasBudget: this._gasBudget,
-      gasPrice: this._gasPrice,
-      gasPayment: this._gasPayment,
+      gasData: this._gasData,
```
Comment (sachushaji) [inline modules/sdk-coin-sui/src/lib/stakingBuilder.ts:46]: Seems to be a breaking change

**Review by sachushaji — COMMENTED** [review]
Kindly call out all the breaking changes in this PR in the commit Footer [format](https://www.conventionalcommits.org/en/v1.0.0/#:~:text=in%20Semantic%20Versioning\).-,BREAKING,-CHANGE%3A%20a))

**Review by byhow — APPROVED** [review]

**Pull request merged**